File: php/tests/ClientTest.php
<?php

namespace Darabonba\Url\Tests;

use PHPUnit\Framework\TestCase;
use Darabonba\Url\Client;

/**
 * @internal
 * @coversNothing
 */
class ClientTest extends TestCase
{
    public function testParseUrl()
    {
        $client = new Client();
        $url    = $client->parseUrl('https://sdk:test@ecs.aliyuncs.com:443/sdk/?api&ok=test#sddd');
        $this->assertEquals([
            'scheme' => 'https',
            'auth'   => [
                'username' => 'sdk',
                'password' => 'test',
            ],
            'host'   => [
                'hostname' => 'ecs.aliyuncs.com',
                'port'     => '443',
            ],
            'path'   => [
                'pathname' => '/sdk/',
                'search'   => 'api&ok=test',
            ],
            'hash'   => 'sddd',
        ], $url->toMap());
    }
}

File: csharp/tests/ClientTest.cs
using AlibabaCloud.TeaUrl;

using Xunit;

namespace tests
{
    public class ClientTest
    {
        [Fact]
        public void TestParseUrl()
        {
            string url = "http://user:pass@host.com:8080/p/a/t/h?query=string#hash";
            var urlModel = UrlClient.ParseUrl(url);
            Assert.Equal("http", urlModel.Scheme);
            Assert.Equal("user", urlModel.Auth.Username);
            Assert.Equal("pass", urlModel.Auth.Password);
            Assert.Equal("host.com", urlModel.Host.Hostname);
            Assert.Equal("8080", urlModel.Host.Port);
            Assert.Equal("/p/a/t/h", urlModel.Path.Pathname);
            Assert.Equal("query=string", urlModel.Path.Search);
            Assert.Equal("hash", urlModel.Hash);

        }
    }
}
